### PR TITLE
Feature/schemas

### PR DESCRIPTION
**Pull request by theborgh**
I hope I did this right (first time using prisma), please let me know.

### CODE REVIEW

File: backend/prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  reportedBugs       Bug[]     @relation("reportedBugs")
+  reportedBugsIds    String[]  @db.ObjectId
+}
+
+model Bug {
+  id                String    @id @default(auto()) @map("_id") @db.ObjectId
+  summary           String?
+  markdown          String
+  priority          Priority
+  status            Status
```
Comment (mehdi-zibout) [inline backend/prisma/schema.prisma:48]: Maybe put a default of unassigned

File: backend/prisma/schema.prisma [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  id            String   @id @default(auto()) @map("_id") @db.ObjectId
+  name          String
+  owner         User     @relation("owner", fields: [ownerUsername], references: [username])
+  ownerUsername String   @db.ObjectId
+  developers    User[]   @relation("developers", fields: [developerIds], references: [username])
+  developerIds  String[] @db.ObjectId
+  bugs          Bug[]
+  bugIds        String[] @db.ObjectId
+  createdAt     DateTime @default(now())
+  lastUpdated   DateTime @default(now())
```
[the remark this answers is not in the document]
Comment (theborgh) [reply backend/prisma/schema.prisma]: thanks, fixed!

File: backend/prisma/schema.prisma [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+model User {
+  username           String    @id @default(auto()) @map("_id") @db.ObjectId
```
[the remark this answers is not in the document]
Comment (theborgh) [reply backend/prisma/schema.prisma]: good point, fixed!

File: backend/prisma/schema.prisma [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  email              String    @unique
+  password           String
+  avatarUrl          String
```
[the remark this answers is not in the document]
Comment (theborgh) [reply backend/prisma/schema.prisma]: good idea, fixed

File: backend/prisma/schema.prisma [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  ownedProjects      Project[] @relation("owner")
+  ownedProjectIds    String[]  @db.ObjectId
+  assignedProjects   Project[] @relation("developers", fields: [assignedProjectIds], references: [id])
+  assignedProjectIds String[]  @db.ObjectId
+  assignedBugs       Bug[]     @relation("assignedBugs")
```
[the remark this answers is not in the document]
Comment (theborgh) [reply backend/prisma/schema.prisma:36]: Adding those fields was giving me problems so I removed assignedBugIds and reportedBugsIds from the schema and I'm going to fetch those by filtering the bugs collection instead

File: backend/prisma/schema.prisma (hunk 1 above)
Comment (theborgh) [reply backend/prisma/schema.prisma]: done, thanks

File: backend/prisma/schema.prisma [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  summary           String?
+  markdown          String
+  priority          Priority
+  status            Status
+  minutesToComplete Int
+  createdAt         DateTime  @default(now())
+  lastUpdated       DateTime  @default(now())
+  reportingUser     User      @relation("reportedBugs", fields: [reportingUserId], references: [username])
+  reportingUserId   String    @db.ObjectId
+  assignedTo        User      @relation("assignedBugs", fields: [assignedToUserId], references: [username])
```
[the remark this answers is not in the document]
Comment (theborgh) [reply backend/prisma/schema.prisma]: good idea, done

File: backend/prisma/schema.prisma (hunk 5 above)
[the remark this answers is not in the document]
Comment (mehdi-zibout) [reply backend/prisma/schema.prisma:36]: I checked prisma docs for one-to-many relations https://www.prisma.io/docs/concepts/components/prisma-schema/relations/one-to-many-relations , it turns out you actually don't need `assignedBugIds`.

**Pull request closed without merging**